Bottom: d87259c4dba294a2524e4b35fa46bd1bb880b11e
Top:    a3d95984d7f6659039e0d9edf45f55ff3b4e67b1
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2012-12-22 16:02:05 +0000

Add a new flag, "connected" / CONNECTEDFLAG, to mark physical connections

XXX: ************

THIS TEXT BELOW IS NOT YET TRUE

This commit reverts the behaviour of the FOUNDFLAG to cover all logical
connections, and introduces a new flag to identify physically connected
objects.


---

diff --git a/src/const.h b/src/const.h
index 3b4dbb5..cc3c450 100644
--- a/src/const.h
+++ b/src/const.h
@@ -214,6 +214,7 @@ Marker used internally to avoid revisiting an object.
 #define EDGE2FLAG               0x4000  /* Padr.Point2 is closer to outside edge */
                                         /* also pinout text for pins is vertical */
 #define VISITFLAG               0x8000  /* marker to avoid re-visiting an object */
+#define CONNECTEDFLAG          0x10000  /* flag like FOUND flag, but used to identify physically connected objects (not rats) */
 /* ---------------------------------------------------------------------------
  * PCB flags
  */
diff --git a/src/strflags.c b/src/strflags.c
index 3699c53..f29da5b 100644
--- a/src/strflags.c
+++ b/src/strflags.c
@@ -108,7 +108,8 @@ static FlagBitsType object_flagbits[] = {
   { LOCKFLAG, N ("lock"), ALL_TYPES },
   { EDGE2FLAG, N ("edge2"), ALL_TYPES },
   { FULLPOLYFLAG, N ("fullpoly"), POLYGON_TYPE},
-  { NOPASTEFLAG, N ("nopaste"), PAD_TYPE }
+  { NOPASTEFLAG, N ("nopaste"), PAD_TYPE },
+  { CONNECTEDFLAG, N ("connected"), ALL_TYPES }
 };
 
 static FlagBitsType pcb_flagbits[] = {
